docs/client.rst

```diff
@@ -72,18 +73,118 @@ WBEM Operation recording
-and can be enabled by setting the
-:attr:`~pywbem.WBEMConnection.operation_recorder` instance variable of the
-:class:`~pywbem.WBEMConnection` object to an operation recorder object,
-i.e. to an object of a subclass of :class:`~pywbem.BaseOperationRecorder`.
-
-Typical usage scenarios for operation recorders are the tracing of WBEM
-operations, or the generation of test cases.
-
+operations that are executed on a connection.
 WBEM Operation recording uses the classes and subclasses defined in
 :ref:`WBEM Operation Recorder`.
 
+This is disabled by default and can be enabled by adding recorders to a
+WBEMConnection with the method
+:meth:`~pywbem.WBEMConnection.add_operation_recorder`.
+
+Typical usage scenarios for various operation recorders are the tracing of WBEM
+operations, tracing of interactions with a WBEM server, or the generation of
+test cases.
+
+Please not that the method of activating operation recorders changed starting
+with pywbem 0.11.0 and the addition of a second recorder.  See
+:meth:`~pywbem.WBEMConnection.add_operation_recorder` for more information.
+
+This adds the recorder defined in the method call to a list of active recorders
+in WBEMConnection. All active recorders are called for each WBEMConnection
+method that sends information to the WBEM server.
+
+A single recorder can be  be disabled with
+:meth:`~pywbem.BaseOperationRecorder.disable` method and enabled with
+:meth:`~pywbem.BaseOperationRecorder.enable` method.
+
+The logger names for the operations and http loggers must be
+defined using  the :class:`~pywbem:`PywbemLoggers` class to
+define usable loggers because the pywbem loggers include logging attributes in
+addition to the standard python logger attributes.  This is done with the
+method :meth:`~pywbem.PywbemLoggers.create_logger` which defines one or more
+loggers from the input parameters.  There is also a convience method
+:meth:`~pywbem.PywbemLoggers.create_loggers` which will define multiple pywbem
+logger definitions from a string definition that could be used with command
+line tools.
+
+The following example shows defining loggers for both http and operations and
+adding activating the LogOperationRecorder in WBEMConnection.
+
+::
+
+    # set the python logger facility to output at the DEBUG level
+    logger.setLevel(logging.DEBUG)
+
+    # Define the parameters of the LogOperationRecorder
+    # using one of the methods in the PywbemLoggers class
+    PywbemLoggers.create_logger('all',
+                                log_dest='file',
+                                log_detail_level='min'
+                                log_filename='test.log')   # define both loggers
+
+    # Create the connection and enable the logger
+    conn = WBEMConnection(...)
+    logging = LogOperationRecorder()
+    add_operations_recorder(LogOperationRecorder())
+    # The LogOperationRecorder is now active and writing logs to stderr.
+    # To enable writing logs
```

Comment:
There are several issues in the example code above:
1. Correct the spelling of `add_operations_recorder()` to `add_operation_recorder()`.
2. `add_operation_recorder()` is a method on class `WbemConnection`, but the example code invokes a global function of that name.
3. It is not clear what the `logging` variable is used for, nor what the `LogOperationRecorder` object it holds is used for (there are two such objects created).
4.  `add_operations_recorder()` defines the logger, but the text below says it is now active.
5. On "to enable writing logs": The code for enabling is missing.

Reply:
1. Corrected spelling
2. fixed add_operation_recorder to conn.add_operation_recorder(LogOperationRecorder()) and dropped logging
3. Dropped the last line since logs are now writing.